src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIPacket.NetStandard.cs

```diff
@@ -45,7 +45,15 @@ async Task ReadFromStreamAsync(SNIPacket packet, SNIAsyncCallback cb, Task<int>
                 cb(packet, error ? TdsEnums.SNI_ERROR : TdsEnums.SNI_SUCCESS);
             }
 
-            Task<int> t = stream.ReadAsync(_data, 0, _capacity, CancellationToken.None);
+            Task<int> t;
+            try
+            {
+                t = stream.ReadAsync(_data, 0, _capacity, CancellationToken.None);
+            }
+            catch (Exception ex)
+            {
+                t = Task.FromException<int>(ex);
+            }
 
             if ((t.Status & TaskStatus.RanToCompletion) != 0)
```

Comment:
The old code `(t.Status & TaskStatus.RanToCompletion) != 0` in `SNIPacket.NetStandard.cs` looks strange, since `TaskStatus` is not a flag, so this will even be true for `TaskStatus.WaitingForActivation`, maybe it should be `t.Status == TaskStatus.RanToCompletion` instead to follow the logic in netcoreapp?

Reply:
Nice catch. Looking at the enum values that'll let fault and cancel through. I'm surprised visual studio doesn't complain about doing bitwise operations on a non-[Flag] enum.